jovo-framework/src/middleware/Handler.ts

```diff
@@ -334,9 +334,51 @@ export class Handler implements Plugin {
         Jovo.prototype.toStateIntent = async function (
             state: string | undefined,
             intent: string,
+        ): Promise<any> {
+            // Check for Component Names inside State
+            if (typeof state !== 'undefined' && this.$components) {
+                const components: string[] = Object.keys(this.$components);
+                const states: string[] = state.split('.');
+                const matched: string[] = states.filter(s => components.includes(s));
+
+                if (states.length > 1 && matched.length > 0) {
+                    throw new JovoError(
+                        `Can not use component names as states. Please rename the following states: ${matched.join(', ')}`,
+                        ErrorCode.ERR,
+                        'jovo-framework',
+                        'The state you want to use contains at least one component name.',
+                        'Rename the listed states to not interfere with them.'
+                    );
+                }
+            }
+
+            const currentState = this.getRootState();
+            if (typeof currentState !== 'undefined' && typeof state !== 'undefined') {
+                let currentStates: string[] = currentState.split('.');
+
+                if (typeof 'state' !== undefined && currentStates[currentStates.length - 1] !== state) {
+                    currentStates = [ ...currentStates, state];
+                }
+
+                state = currentStates.join('.');
+            }
+
+            return this.internalToStateIntent(state, intent);
+        };
+
+        /**
+         * Jumps to state intent without any validation
+         * @private
+         * @param {string} state name of state
+         * @param {string} intent name of intent
+         */
+        Jovo.prototype.internalToStateIntent = async function (
```

Comment:
Is it really necessary to have both `internalToStateIntent` and `toStateIntent`. Can't we just merge them?

Reply:
We could, the only difference is the added validation on the new `toStateIntent`, `internalToStateIntent` is just the old method. But then there must be a way to toggle the validation using a parameter (which I don't like, because we should enforce the validation and not make it easy to disable) or a instance variable similar to `this.triggeredToIntent`

Reply:
I've added the `this.triggerStateValidation` property for now to determine if the state should be validated. But I can change it to whatever you prefer naming-wise.

Reply:
I'd actually prefer using a parameter where we set the default value to `true`. It's more straightforward and is less prone to errors.

For example, with the current implementation, the validation wouldn't be run if we have the following scenario:

`this.delegate()` -> sets `triggerStateValidation` to false. -> in the same request&response, the component uses `this.toStateIntent()` -> since `triggerStateValidation` was set to false, the validation won't be triggered.

@aswetlow what's your take on it?

Reply:
Oh, didn't catch that, im sorry! Fixed by resetting the `triggerStateValidation` back to true before returning. Personally I'm not a huge fan of changing this to a parameter, because anyone who does't pay much attention might messing up the whole flow by unintentionally leaving/entering a component. The validation flag/toggle should remain internal use only, at least in my opinion.

Reply:
Hey!
I think I would prefer Kaan's approach with the parameter. It makes the code a little more manageable and clearer to use (no additional variable in the Jovo class)

Great PR btw!

Reply:
Replaced it with the `validate` param.